builder/project.py

```diff
@@ -547,22 +562,20 @@ def _project_from_path(path='.', name_hint=None):
                           project_config_file, e)
                     sys.exit(1)
 
-                if not project_config.get('name', None):
-                    project_config['name'] = name_hint if name_hint else os.path.dirname(
-                        os.getcwd())
-                print('    Found project: {} at {}'.format(
-                    project_config['name'], path))
-                project = Project._create_project(
-                    **project_config, path=path)
-                return Project._cache_project(project)
+                if name_hint == None or project_config.get('name', None) == name_hint:
```

Comment:
Previously, this and the path assignment below were leading every project to just return the root project as its path, creating the infinite recursion